src/VstsSyncMigrator.Core/Execution/OMatics/RepoOMatic.cs

```diff
@@ -55,6 +55,13 @@ public int FixExternalLinks(WorkItem targetWorkItem, WorkItemStoreContext target
                     ExternalLink el = (ExternalLink)l;
 
                     GitRepositoryInfo sourceRepoInfo = GitRepositoryInfo.Create(el, sourceRepos, migrationEngine, sourceWorkItem?.Project?.Name);
+                    
+                    // if sourceRepo is null ignore this link and keep processing further links
+                    if (sourceRepoInfo == null) 
+                    {
+                        continue;
+                    }
+
                     // if repo was not found in source project, try to find it by repoId in the whole project collection
                     if (sourceRepoInfo.GitRepo == null)
```

Comment:
The null check should be placed here in combination with the current check, since it is trying to find the repo again in allRepos list otherwise the cross-project linking wont work. And the whole sourceRepoInfo null check can be placed after this if block.